File: PART3/ch02/java-kotlin/src/main/java/org/example/ex09/Exam09.java
package org.example.ex09;

public class Exam09 {


    public Exam09(StoreUser storeUser){
        // service logic

        if(storeUser.getRole().equals("MASTER")){
            // master
        }
        else if(storeUser.getRole().equals("ADMIN")){
            // admin
        }
        else if(storeUser.getRole().equals("USER")){
            // user
        }
        else {
            // default
        }

        var role = "";
        switch (storeUser.getRole()){
            case "MASTER" :
            case "ADMIN" :
                role = "MASTER";
            break;
            case "USER" :

            break;
            default :
                // default ??
            break;
        }

        try{

        }catch (Exception e){
            var result = "";
            if(e instanceof NullPointerException){

            }
            else if(e instanceof NumberFormatException){

            }

        }
    }

    public static void main(String[] args){
        new Exam09(new StoreUser());
    }
}


class StoreUser{
    private String name;
    private String role;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }
}

File: PART3/ch01/kotlin-example/kotlin-example/src/main/kotlin/ex09/Exam09.kt
package ex09

import java.lang.NullPointerException
import java.lang.RuntimeException

fun main(){


    val result = when(""){
        "" -> {
            ""
        }
        "MASTER","ADMIN" -> {
            "master"
        }
        "USER" -> {
            "user"
        }
        else -> {
            "default"
        }
    }

    var any : Any = "";
    var exception = RuntimeException()
    when(exception){
        is NullPointerException -> {

        }
        is RuntimeException -> {

        }
    }

    var number = 13


    when(val n = number % 2){
        0 -> {
            println(n)
        }
        else ->{
            println(n)
        }
    }

    val r = when{
        number % 2 == 0 ->{
            100
        }
        else -> {
            200
        }
    }
    println(r)

}